src/exceptions/clientException.ts

```diff
@@ -10,15 +12,13 @@ export class ClientException extends Exception {
     404: 422,
   };
 
-  constructor(serviceName: string, originalStatusCode?: number, error?: any) {
+  constructor(serviceName: string, originalStatusCode?: number, details?: any) {
     super(
       'Dependent service returned error',
       ClientException.convertStatusCode(originalStatusCode),
-      {
-        error,
```

Comment:
The issue here is that we're wrapping the user defined details in the `error` property and there's no way for the user to know unless they look at this code here

Reply:
Yeah, I agree that is still not good. I was already "bug" fixing some code to get it from the `error` property. So I 👍 your change.

Are we dropping the serviceName from the exception that will get logged though?

Reply:
No noo that wasn't the intent. But good point, I made a test and the `serviceName` is no longer visible on the API response (it's still logged). First, do we want it there? If yes I see two options - adding it to the details which is a bit fragile - as it's `details?: any`. Or appending it to the title of the message

Reply:
I can only say it is useful (e.g. people know whom to contact) when the error response from a service (e.g. with 503) includes the service name that failed underneath.

Reply:
(I did the easier fix, but let me know if you prefer the other)

Reply:
I was thinking it's internal details of the service and the API caller doesn't need to necessarily care/know what other services we call. For internal users, I agree it's a big plus if they can do something with this information. All good then!